1_The_Physical/Level_2_Stateful_Guess_Point/Shared/Student/src/domain/student/value-objects/last-name.ts

```diff
@@ -0,0 +1,64 @@
+import { Result } from "../../../shared/result";
+import { ValueObject } from "../../core/value-object";
+
+interface LastNameProps {
+  value: string;
+}
+
+export interface LastNameValidationError {
+  min?: string;
+  max?: string;
+  letters?: string;
+  required?: string;
+}
+
+export class LastName extends ValueObject<LastNameProps> {
```

Comment:
This is the same as `FistName`. I would consider consolating them.

Reply:
I completely agree with you, I had already prepared for this with this [commit](https://github.com/Lokicoule/the-software-essentialist/pull/1/commits/ccb074d8b50b0938362660d657ca164fbbcb499d).

But after rereading the instructions and this specific part:
```
🔘 There are no duplicated instances of validation logic anywhere. Everything is in its rightful place.
```

I understood that Khalil wanted us to keep the validation logic specific to each value object. That's how I interpreted it, and I also see the value in it. In the future, if we need to add rules to one or the other value object, we won't risk creating any side effects. Conversely, if we need to add a new common rule, we'll have to replicate it in both value objects.

Reply:
>  In the future, if we need to add rules to one or the other value object, we won't risk creating any side effects. Conversely, if we need to add a new common rule, we'll have to replicate it in both value objects.

Well, it depends on the implementation. For example, you could implement a validator class and then use it in both value objects then the actual validation logic would be written only once but used in multiple places. Even right now you have the validation parametrized, so that the parameters tell what rules need to be applied to the value objects. Also, you have tests that would catch side effects.